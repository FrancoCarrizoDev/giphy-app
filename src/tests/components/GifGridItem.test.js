import { shallow } from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem'
import React from 'react';

describe('Tets in <GifGridItem/>', () => {


    const title = 'Un t√≠tulo'
    const url = 'https://google.com'
    const wrapper = shallow(<GifGridItem title={title} url={url} />)


    test('should the component correctly', () => {
        
        expect(wrapper).toMatchSnapshot()
    })


    test('should have an parragrapth with', () => {
        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(title)
    })
    
    test('should have the img equal url and alt to theys props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    })

    test('should have a animate_Css', () => {
        const div = wrapper.find('div')
        const className = div.prop('className')
        expect(className.includes('animate__fadeIn')).toBe(true)
    })
    

})
