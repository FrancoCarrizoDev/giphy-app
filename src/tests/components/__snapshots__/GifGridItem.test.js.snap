// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tets in <GifGridItem/> should the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://google.com"
  />
  <p>
     
    Un título
     
  </p>
</div>
`;
